### PR TITLE
fix(lightning): update transfer UI

### LINKED ISSUE #1288: [Bug]: In-flight HTLC
### Describe the bug

My channel got force closed while there was an in-flight HTLC. Though the closing transaction has many confirmations, the payment that was in-flight is still showing as pending under the spending balance, and the Lightning suggestion card is still showing "transferring".

IMG_2230
IMG_2231
IMG_2232
IMG_2233

### Screenshots / Recording

[bitkit_ldk_logs_1696301903578.zip](https://github.com/synonymdev/bitkit/files/12788045/bitkit_ldk_logs_1696301903578.zip)

### Operating system

iOS 17

### Bitkit version

72

### Log output

_No response_

### LINKED ISSUE #1335: [Bug]: Activity update
### Describe the bug

The chan closes don't show in activity

Screenshot 2023-10-17 at 11 02 18

https://synonymworkspace.slack.com/archives/C01H106JRQB/p1697533238125939?thread_ts=1697529943.255019&cid=C01H106JRQB

### Screenshots / Recording

_No response_

### Operating system

iOS

### Bitkit version

v82

### Log output

_No response_

### LINKED ISSUE #1484: [Bug]: Channel closing UX
I closed a channel locally, and my reserve was already sweeped to my on-chain wallet, but the sweep is still shown as pending. (ps: channel was opened via the Bitfinex and Blocktank integration).
image
image
image
image

I later opened and closed another channel, and it happened again (fortunately, the 4900 sats sweeped from the previous channel is no longer appearing as "pending"):
IMG_3C36AE238235-1

### PR DESCRIPTION
**Pull request by pwltr**
### Description

Updates the UI to better show transfers from and to spending, also contains most of the design updates from v46 related to transfers. I moved some state related to transfers to its own field in `wallet.transfers`, some of this is duplicated but this makes it more explicit and way easier to work with.

### Linked Issues/Tasks

Add any links to GitHub issues or Asana tasks that are relevant to this pull request.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (improving code without creating new functionality)

### Tests

- [ ] Detox test
- [x] Unit test
- [ ] No test

### Screenshot / Video

channel open:

https://github.com/synonymdev/bitkit/assets/8538369/e798f87a-b2c1-45a0-b0e8-ea7b3e1cf07e

channel close:

https://github.com/synonymdev/bitkit/assets/8538369/b01c3e5a-8413-410c-ae4a-58147a5e029a

### CODE REVIEW

File: src/utils/wallet/index.ts [hunk 1]
```diff
@@ -2305,12 +2304,6 @@ export const switchNetwork = async (
 		updateWallet({ selectedNetwork: originalNetwork });
 		return err(response.error.message);
 	}
-	// Start wallet services with the newly selected network.
-	await startWalletServices({
-		selectedNetwork,
-	});
```
Comment (pwltr) [inline src/utils/wallet/index.ts]: @coreyphillips had to move this out of the function because a circular dependency causing weird issues.

Comment (coreyphillips) [reply src/utils/wallet/index.ts]: We need this to switch over everything to the new network. Where's the circular dependency?

Comment (pwltr) [reply src/utils/wallet/index.ts]: The line has been moved [here](https://github.com/synonymdev/bitkit/blob/d558994693f09d6315a530eaf667dd481d6d55ad/src/screens/Settings/Bitcoin/BitcoinNetworkSelection.tsx#L33) so it should still behave as expected. I wasn't able to track down the circular import. There also was another one causing `HEADER_HEIGHT` not to be applied which is why I duplicated the variable to where it was imported [here](https://github.com/synonymdev/bitkit/blob/d558994693f09d6315a530eaf667dd481d6d55ad/src/screens/Wallets/index.tsx#L42). Not sure why these are popping up all of a sudden, I know we have a ignore rule for the warnings that would show this but wasn't an issue until now.

File: src/store/actions/wallet.ts [hunk 2]
```diff
@@ -296,6 +301,78 @@ export const injectFakeTransaction = ({
...
+	const pendingTransfers = transfers.filter((t) => {
+		return (
+			t.type !== ETransferType.open && t.status === ETransferStatus.pending
+		);
+	}) as TTransferToSavings[];
+
+	pendingTransfers.forEach((transfer) => {
+		const tx = transactions[transfer.txId];
+		if (transfer.confirmations <= 6) {
+			const confs = tx.height === -1 ? 1 : headerHeight - tx.height + 1;
```
Comment (coreyphillips) [inline src/store/actions/wallet.ts]:
```suggestion
			const confs = tx.height < 1 ? 0 : headerHeight - tx.height + 1;
```
Seeing a confirmation miscalculation. Possibly from this snippet.
Screenshot 2024-02-10 at 12 48 03 PM

**Review by coreyphillips — CHANGES_REQUESTED** [review]

**Comment by coreyphillips** [comment]
Other than that, everything LGTM!

**Review by limpbrains — APPROVED** [review]

**Pull request closed without merging**